addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/handler/UniFiBaseThingHandler.java

```diff
@@ -0,0 +1,133 @@
+import org.openhab.binding.unifi.internal.api.model.UniFiController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Matthew Bowman - Initial contribution
+ *
+ * @param <E> entity - the UniFi entity class used by this thing handler
+ * @param <C> config - the UniFi config class used by this thing handler
+ */
+@NonNullByDefault
+public abstract class UniFiBaseThingHandler<E, C> extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(UniFiBaseThingHandler.class);
+
+    public UniFiBaseThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public final void initialize() {
+        Bridge bridge = getBridge();
+        if (bridge == null || bridge.getHandler() == null
+                || !(bridge.getHandler() instanceof UniFiControllerThingHandler)) {
+            updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "You must choose a UniFi Controller for this thing.");
+            return;
+        }
+        if (bridge.getStatus() == OFFLINE) {
+            updateStatus(OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "The UniFi Controller is currently offline.");
+        }
+        // mgb: derive the config class from the generic type
+        Class<?> clazz = (Class<?>) (((ParameterizedType) getClass().getGenericSuperclass())
+                .getActualTypeArguments()[1]);
+        C config = (C) getConfigAs(clazz);
+        initialize(config);
+    }
+
+    /**
+     * Utility method to access the {@link UniFiController} instance associated with this thing.
+     *
+     * @return
+     */
+    @SuppressWarnings("null")
+    private final @Nullable UniFiController getController() {
+        Bridge bridge = getBridge();
+        if (bridge != null && bridge.getHandler() != null
+                && (bridge.getHandler() instanceof UniFiControllerThingHandler)) {
+            return ((UniFiControllerThingHandler) bridge.getHandler()).getController();
+        }
+        return null;
+    }
+
+    @Override
+    public final void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Handling command = {} for channel = {}", command, channelUID);
+        // mgb: only handle commands if we're ONLINE
+        if (getThing().getStatus() == ONLINE) {
+            UniFiController controller = getController();
+            if (controller != null) {
+                E entity = getEntity(controller);
+                if (entity != null) {
+                    if (command == REFRESH) {
+                        refreshChannel(entity, channelUID);
+                    } else {
+                        try {
+                            handleCommand(entity, channelUID, command);
+                        } catch (UniFiException e) {
+                            logger.error("Unexpected error handling command = {} for channel = {} : {}", command,
```

Comment:
Don't log to error, it's reserved within openHAB for system wide errors. Make it warn or debug.